### ISSUE TITLE
Bucket selector script is not applied to empty histogram buckets

### ISSUE THREAD

**Issue by AndreKR** [elastic/elasticsearch#15471]
Consider this dataset:

```
POST _bulk
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 1, "b": 1}
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 1, "b": 2}
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 3, "b": 1}
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 3, "b": 3}
```

Run this aggregation on it:
```
POST aggtest/default/_search?size=0
{
  "aggs": {
    "by_a": {
      "histogram": {
        "field": "a",
        "interval": 1
      },
      "aggs": {
        "by_b": {
          "range": {
            "field": "b",
            "ranges": [
              { "from": 1, "to": 2 },
              { "from": 2, "to": 3 },
              { "from": 3, "to": 4 }
            ]
          },
          "aggs": {
            "the_filter": {
              "bucket_selector": {
                "buckets_path": {
                  "the_doc_count": "_count"
                },
                "script": {
                  "inline": "the_doc_count > 0",
                  "lang": "expression"
                }
              }
            }
          }
        }
      }
    }
  }
}
```

This is the result. Note how the empty buckets of `by_b` are not removed by the script when the parent bucket of `by_a` is empty.

![Result](https://cloud.githubusercontent.com/assets/1188538/11832268/fb045e06-a3b6-11e5-8623-677fe47f79c3.png)

**Comment by clintongormley** [comment]
@colings86 could you take a look please?

**Comment by colings86** [comment]
@AndreKR thanks for raising this, I have merged a PR with the fix and it should be available from version 2.2. In the meantime a workaround could be to set min_doc_count on the histogram to 1.
